downstairs/src/main.rs

```diff
@@ -744,7 +777,7 @@ where
                                 ds.promote_to_active(
                                     uuid,
                                     another_upstairs_active_tx.clone()
-                                ).await;
+                                ).await?;
```

Comment:
If we fail in promote to active, it's because we have failed to re-open all extents, which means things are bad.  We will also exit `proc()` with an error and then the whole connection to the upstairs will close and re-open, which should send at least some message to upstairs that things are wrong.

Are you suggesting adding a "PromoteToActiveFailed" message to the protocol here?  It might give the upstairs a little more info, but the upstairs is still going to have to send this downstairs through re-connection anyway.

Reply:
>  We will also exit proc() with an error and then the whole connection to the upstairs will close and re-open, which should send at least some message to upstairs that things are wrong.

Let's not add a new "PromoteToActiveFailed" message, the upstairs should notice the disconnects and deal with it accordingly.